csrc/codegen.cpp

```diff
@@ -427,6 +427,9 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
           if (has_parallel_welford) {
             smem_buf_size_ss << " * 3";
           }
+          if (kernel_summary.all_block_reductions_are_warp_reduction) {
+            smem_buf_size_ss << " / 32";
+          }
```

Comment:
This is the part where I was asking about the padded threads that doesn't require smem buffer.